overlord/ifacestate/handlers.go

```diff
@@ -85,6 +85,21 @@ func (m *InterfaceManager) doSetupProfiles(task *state.Task, tomb *tomb.Tomb) er
 		return err
 	}
 
+	var corePhase2 bool
+	if err := task.Get("core-phase-2", &corePhase2); err != nil && err != state.ErrNoState {
+		return err
+	}
+	if corePhase2 {
+		if snapInfo.Type != snap.TypeOS {
```

Comment:
Wait are you sure there is nothing to do? Would this skip all of the setup-profiles that happen to run before before snapd restarts?